app-modules/meeting-center/src/Providers/MeetingCenterServiceProvider.php

```diff
@@ -26,6 +28,16 @@ public function boot(): void
             'calendar_event' => CalendarEvent::class,
         ]);
 
+        $this->callAfterResolving(Schedule::class, function (Schedule $schedule) {
+            // TODO Ensure we are locking entities that have already been picked up for processing to avoid overlap
+            Calendar::all()
```

Comment:
Let's use `cursor()` here instead of `all()`. In the event that we have a large number of records, this will reduce memory usage as it doesn't load every record into a collection all at once.